code/game/objects/machinery/autodoc.dm

```diff
@@ -674,6 +676,14 @@
 		playsound(loc,'sound/machines/buzz-two.ogg', 25, 1)
 		return
 	if(occupant)
+		if(forceeject)
+			if(!surgery)
+				visible_message("\The [src] is destroyed, ejecting [occupant] and showering them in debris.")
+				occupant.take_limb_damage(rand(10,20),rand(10,20))
+				go_out(AUTODOC_NOTICE_FORCE_EJECT)
+			visible_message("\The [src] malfunctions as it is destroyed mid-surgery, ejecting [occupant] with surgical wounds and showering them in debris.")
```

Comment:
Does a return fit here? Else there will be two messages and two `go_out` being called.

Reply:
It didn't cause gameplay issues but a runtime where it tried to damage null.limb
Oops